### PR TITLE
DOC: add input/output/parameter help text

### PR DESCRIPTION
**Pull request by gregcaporaso**

### CODE REVIEW

File: q2_feature_table/plugin_setup.py [hunk 1]
```diff
@@ -47,6 +53,11 @@
     inputs={'table': FeatureTable[Frequency | RelativeFrequency]},
     parameters={},
     outputs=[('presence_absence_table', FeatureTable[PresenceAbsence])],
+    input_descriptions={'table': 'The feature table to be converted.'},
```
Comment (jairideout) [inline q2_feature_table/plugin_setup.py]: *The feature table to be converted into presence/absence abundances.*

File: q2_feature_table/plugin_setup.py [hunk 2]
```diff
@@ -47,6 +53,11 @@
...
+    parameter_descriptions={},
+    output_descriptions={
+        'presence_absence_table': 'The resulting feature table.'
```
Comment (jairideout) [inline q2_feature_table/plugin_setup.py]: *The resulting presence/absence feature table.*

File: q2_feature_table/plugin_setup.py [hunk 3]
```diff
@@ -59,6 +70,11 @@
     outputs=[
         ('relative_frequency_table',
          FeatureTable[RelativeFrequency] % Properties('uniform-sampling'))],
+    input_descriptions={'table': 'The feature table to be converted.'},
```
Comment (jairideout) [inline q2_feature_table/plugin_setup.py]: *The feature table to be converted into relative frequencies.*

File: q2_feature_table/plugin_setup.py [hunk 4]
```diff
@@ -59,6 +70,11 @@
...
+    parameter_descriptions={},
+    output_descriptions={
+        'relative_frequency_table': 'The resulting feature table.'
```
Comment (jairideout) [inline q2_feature_table/plugin_setup.py]: *The resulting relative frequency feature table.*

File: q2_feature_table/plugin_setup.py [hunk 5]
```diff
@@ -72,6 +88,12 @@
     parameters={},
     outputs=[
         ('merged_table', FeatureTable[Frequency])],
+    input_descriptions={
+        'table1': 'The first feature table to be merged.',
+        'table2': 'The second feature table to be merged.',
+    },
+    parameter_descriptions={},
+    output_descriptions={'merged_table': 'The resulting feature table.'},
```
Comment (jairideout) [inline q2_feature_table/plugin_setup.py]: *The resulting merged feature table containing all features from `table1` and `table2`.*

File: q2_feature_table/plugin_setup.py [hunk 6]
```diff
@@ -85,6 +107,14 @@
     parameters={},
     outputs=[
         ('merged_data', FeatureData[Sequence])],
+    input_descriptions={
+        'data1': 'The first collection of feature sequences to be merged.',
+        'data2': 'The second collection of feature sequences to be merged.',
+    },
+    parameter_descriptions={},
+    output_descriptions={
+        'merged_data': 'The resulting collection of feature sequences.'
```
Comment (jairideout) [inline q2_feature_table/plugin_setup.py]: *The resulting collection of feature sequences containing all feature sequences from `data1` and `data2`.*

File: q2_feature_table/plugin_setup.py [hunk 7]
```diff
@@ -101,6 +131,14 @@
     parameters={},
     outputs=[
         ('merged_data', FeatureData[Taxonomy])],
+    input_descriptions={
+        'data1': 'The first collection of feature taxonomies to be merged.',
+        'data2': 'The second collection of feature taxonomies to be merged.',
+    },
+    parameter_descriptions={},
+    output_descriptions={
+        'merged_data': 'The resulting collection of feature taxonomies.'
```
Comment (jairideout) [inline q2_feature_table/plugin_setup.py]: *The resulting collection of feature taxonomies containing all feature taxonomies from `data1` and `data2`.*

File: q2_feature_table/plugin_setup.py [hunk 8]
```diff
@@ -124,13 +157,36 @@
...
+        'max_features': ('The maximum number of features that a sample can '
+                          'have to be retained.'),
+        'sample_metadata': 'Sample metadata used in conjuction with `where` '
+                           'parameter to select samples to retain.',
+        'where': 'SQLite WHERE clause specifying sample metadata criteria '
+                 'that must be met to be included in the filtered feature '
+                 'table. If not provided, all samples in `sample_metadata` '
+                 'that are also in the feature table will be retained.'
+    },
+    output_descriptions={'filtered_table': 'The resulting feature table.'},
```
Comment (jairideout) [inline q2_feature_table/plugin_setup.py]: *The resulting feature table filtered by sample.*

File: q2_feature_table/plugin_setup.py [hunk 9]
```diff
@@ -124,13 +157,36 @@
...
     name="Filter samples from table.",
     description="Filter samples from table based on frequency and/or "
                 "metadata. Any features with a frequency of zero after sample "
                 "filtering will also be removed. If no value(s) are provided "
                 "for max_frequency or max_features, they will default to "
```
Comment (jairideout) [inline q2_feature_table/plugin_setup.py]: Can you move these default descriptions to their corresponding param help text?

File: q2_feature_table/plugin_setup.py [hunk 10]
```diff
@@ -143,20 +199,45 @@
...
+        'max_samples': ('The maximum number of samples that a feature can '
+                          'be observed in to be retained.'),
+        'feature_metadata': 'Feature metadata used in conjuction with `where` '
+                           'parameter to select features to retain.',
+        'where': 'SQLite WHERE clause specifying feature metadata criteria '
+                 'that must be met to be included in the filtered feature '
+                 'table. If not provided, all features in `feature_metadata` '
+                 'that are also in the feature table will be retained.'
+    },
+    output_descriptions={'filtered_table': 'The resulting feature table.'},
```
Comment (jairideout) [inline q2_feature_table/plugin_setup.py]: *The resulting feature table filtered by feature.*

File: q2_feature_table/plugin_setup.py [hunk 11]
```diff
@@ -143,20 +199,45 @@
...
     name="Filter features from table.",
     description="Filter features from table based on frequency and/or "
                 "metadata. Any samples with a frequency of zero after feature "
                 "filtering will also be removed. If no value(s) are provided "
                 "for max_frequency and/or max_samples, they will default to "
```
Comment (jairideout) [inline q2_feature_table/plugin_setup.py]: same comment here re: moving default descriptions to param help text

**Comment by jairideout** [comment]
Note that travis is failing due to flake8 errors

File: q2_feature_table/plugin_setup.py [hunk 12]
```diff
@@ -34,6 +34,12 @@
     parameters={'sampling_depth': Int},
     outputs=[('rarefied_table',
               FeatureTable[Frequency] % Properties('uniform-sampling'))],
+    input_descriptions={'table': 'The feature table to be rarefied.'},
+    parameter_descriptions={
+        'sampling_depth': ('The total frequency that each sample should be '
```
Comment (gregcaporaso) [inline q2_feature_table/plugin_setup.py]: I bumped it up to here only.

**Comment by gregcaporaso** [comment]
Addressed your comments @jairideout. Thanks for the review!

**Comment by jairideout** [comment]
Thanks @gregcaporaso!

**Pull request merged**